src/main/java/seedu/address/logic/parser/SearchStudentCommandParser.java

```diff
@@ -0,0 +1,86 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULECODE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENTID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TUTORIALCLASS;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.SearchStudentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.module.ModuleContainsKeywordPredicate;
+import seedu.address.model.module.TutorialContainsKeywordPredicate;
+import seedu.address.model.person.EmailContainsKeywordPredicate;
+import seedu.address.model.person.NameContainsKeywordPredicate;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.StudentIdContainsKeywordPredicate;
+
+/**
+ * Parses input arguments and creates a new SearchStudentCommand object
+ */
+public class SearchStudentCommandParser implements Parser<SearchStudentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SearchStudentCommand
+     * and returns a SearchStudentCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SearchStudentCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(
+                        args, PREFIX_NAME, PREFIX_STUDENTID, PREFIX_EMAIL, PREFIX_MODULECODE, PREFIX_TUTORIALCLASS);
+
+        if (!hasOnlyOnePrefix(argMultimap, PREFIX_NAME, PREFIX_STUDENTID, PREFIX_EMAIL,
+                PREFIX_MODULECODE, PREFIX_TUTORIALCLASS)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchStudentCommand.MESSAGE_USAGE));
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_EMAIL, PREFIX_STUDENTID,
+                PREFIX_MODULECODE, PREFIX_TUTORIALCLASS);
+
+        Predicate<Person> predicate = null;
+
+        if (argMultimap.getValue(PREFIX_TUTORIALCLASS).isPresent()) {
```

Comment:
if possible, can use switch + cases instead of so many if statements! 
e.g. 
for (Prefix prefix : prefixes) {
    if (argMultimap.getValue(prefix).isPresent()) {
        String keyword = argMultimap.getValue(prefix).get();
        switch (prefix) {
            case PREFIX_TUTORIALCLASS:
            ....
            case PREFIX_MODULECODE:

Reply:
Thanks for the suggestion @qinxutan! However, I feel as though the switch statement being nested in the for loop might make the code a bit harder to follow. I do agree that it will be better to reduce repeated code. Since it's functional for now, maybe we could merge this PR first and I can update it in a separate future PR 😸

Reply:
ok!! ill go ahead and merge it first :-)